src/type/__tests__/validation-test.js

```diff
@@ -991,21 +1079,47 @@ describe('Type System: Interface fields must have output types', () => {
 
   it('rejects an empty Interface field type', () => {
     const schema = schemaWithInterfaceFieldOfType(undefined);
-    expect(validateSchema(schema)).to.deep.include({
-      message:
-        'The type of BadInterface.badField must be Output Type but got: undefined.',
-    });
+    expect(validateSchema(schema)).to.deep.equal([
+      {
+        message:
+          'The type of BadInterface.badField must be Output Type but got: undefined.',
+      },
+      {
+        message:
+          'The type of BadImplementing.badField must be Output Type but got: undefined.',
+      },
+    ]);
   });
 
   notOutputTypes.forEach(type => {
     it(`rejects a non-output type as an Interface field type: ${type}`, () => {
       const schema = schemaWithInterfaceFieldOfType(type);
-      expect(validateSchema(schema)).to.deep.include({
-        message: `The type of BadInterface.badField must be Output Type but got: ${type}.`,
-      });
+      expect(validateSchema(schema)).to.deep.equal([
+        {
+          message: `The type of BadInterface.badField must be Output Type but got: ${type}.`,
+        },
+        {
+          message: `The type of BadImplementing.badField must be Output Type but got: ${type}.`,
+        },
+      ]);
     });
   });
 
+  it('rejects a non-type value as an Interface field type', () => {
+    const schema = schemaWithInterfaceFieldOfType(Number);
+    expect(validateSchema(schema)).to.deep.equal([
+      {
+        message: `The type of BadInterface.badField must be Output Type but got: ${Number}.`,
+      },
+      {
+        message: `Expected GraphQL named type but got: ${Number}.`,
```

Comment:
This one feels low value and duplicative of the error below it, don't you think?

Reply:
@leebyron Agree 👍 
However I don't see how `validate` can distinguish beetween those two cases:
```js
validateSchema(new GraphQLSchema({
  queryType: Number,
}));
```
```js
validateSchema(new GraphQLSchema({
  queryType: new GraphQLObjectType({
    name: 'Foo',
    fields: { bar: { type: GraphQLString } },
  }),
  types: [Number],
}));
```

In the first case `Expected GraphQL named type but got: ${Number}.` is redundant but in second one it's the only error that would be reported.
Can you suggest how to solve this?